sympy/simplify/simplify.py

```diff
@@ -422,6 +422,33 @@ def signsimp(expr, evaluate=None):
         e = e.replace(lambda x: x.is_Mul and -(-x) != x, lambda x: -(-x))
     return e
 
+def simplify_nested_floor_ceil(expr):
+    """
+    Simplifies nested floor divisions (//) and ceiling operations in a SymPy expression.
+
+    Args:
+        expr: A SymPy expression.
+
+    Returns:
+        A simplified SymPy expression with nested floor and ceiling operations collapsed.
+    """
+    if isinstance(expr, (floor, ceiling)):
+        arg = expr.args[0]
+
+        if isinstance(arg, Expr):
+            numerator, denominator = arg.as_numer_denom()
+
+            if denominator.is_integer and denominator.is_positive:
+                if isinstance(numerator, (floor, ceiling)):
+                    inner_function = numerator.args[0]
+
+                    if isinstance(expr, floor):
+                        return floor(inner_function / denominator)
+
+                    elif isinstance(expr, ceiling):
+                        return ceiling(inner_function / denominator)
+
+    return expr
```

Comment:
This function is far too specific. No one would ever want to call it. Something like this would only be useful if it was more general. At minimum it should recurse the expression tree rather than checking if the head is a floor.

It is also too specific to be used as part of the `simplify` function.

Where it might potentially make sense is in `floor.eval` so that it happens automatically when calling `floor` but I am also not sure there either. The rules for something like `floor.eval` should be kept simple but applying it there has the advantage that it is implicitly recursive.

Reply:
Thanks @oscarbenjamin for your time and suggestions. I didn't make it recursive because the call to simplify with the original expression was implicitly invoking simplify_nested_floor_ceil on sub-expressions (making all test cases pass), but I definitely see your point having a general recursive function.

I can look into `floor.eval` and see how this functionality can be added with indeed maintaining its simplicity intact as much as possible .

Reply:
@zq1997 Quick question, does this theorem also apply to a combination of floor and ceiling wrapped calls or only when they are all floor or all ceiling?

Reply:
I guess not, since `floor(floor(x)/2)` and `floor(ceiling(x)/2)` disproves it, when `x = 3.9`.

Conversely, `ceiling(floor(x)/2)` and `ceiling(ceiling(x)/2)` disproves it, when `x = 4.1`.

Reply:
@oscarbenjamin, I tried implementing a fix both recursively and iteratively, but given the edge cases above, it should only collapse the nested expression when they are all of the same type (all floor or all ceiling), which would impact the simplicity of the implementation.

The simplest solution I could come up with is this iterative implementation:

```python
def collapse_nested_rounds(expr, round_function):
    numerator = expr
    denominator_final = 1
    while isinstance(expr, round_function):
        arg = expr.args[0]

        if isinstance(arg, Expr):
            numerator, denominator = arg.as_numer_denom()
            if denominator.is_integer and denominator.is_positive:
                denominator_final *= denominator
                expr = numerator

    return numerator / denominator_final
```

This function is called in the `eval` method of the `RoundFunction` class as follows:

```python
collapsed_arg = collapse_nested_rounds(arg, cls)
if collapsed_arg != arg:
    return cls(collapsed_arg)
```

But again, this collapses the above cases as well to the outermost rounding function which should not be expected. 

Any suggestions on this one?